game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/MockGameData.java

```diff
@@ -29,6 +33,18 @@ public GameData build() {
     return gameData;
   }
 
+  public MockGameData withAlliedRelationship(
+      final GamePlayer player1, final GamePlayer player2, final boolean value) {
+    when(relationshipTracker.isAllied(player1, player2)).thenReturn(value);
+    return this;
+  }
+
+  public MockGameData withWarRelationship(
```

Comment:
At a first glance this is a bit confusing to have a boolean parameter that negates the whole thing. For consideration, boolean parameters automatically violate SRP, the method by definition then does multiple things (so long as the parameter is used).

Perhaps the underlying code should get the relationship type instead of using the 'isAllied' and 'isAtWar' methods?

In that case you could have `withRelationship(Player1, Player2, RelationshipType)`

Reply:
I would have to make the mocks lenient.  Because making a player be at war with another requires that the allied be false and the war true.  But both of those mocked calls will not be executed during the test and the test will fail.

Also, this is really just a wrapper around the `isAtWar` and `isAllied` methods to make give it a fluent api.  So, if you mocked the relationship outside of this wrapper, you'd still have the same problems you mentioned.